tests/unit/src/cucumber-runner.spec.js

```diff
@@ -0,0 +1,79 @@
+const { buildArgs, normalizeFormat } = require('../../../src/cucumber-runner');
+
+describe('buildArgs', () => {
+  const cucumberBin = '/usr/local/bin/cucumber';
+
+  it('should build correct arguments with basic configuration', () => {
+    const runCfg = {
+      sauce: {
+        metadata: {},
+      },
+      projectPath: '/project',
+      assetsDir: '/project/assets',
+      suite: {
+        options: {
+          paths: ['features/test.feature'],
+          formatOptions: {
+            myOption: 'test',
+          },
+        },
+      },
+    };
+
+    const result = buildArgs(runCfg, cucumberBin);
+
+    expect(result).toEqual([
+      cucumberBin,
+      '/project/features/test.feature',
+      '--force-exit',
+      '--require-module',
+      'ts-node/register',
+      '--format',
+      '"@saucelabs/cucumber-reporter":"cucumber.log"',
+      '--format-options',
+      '{"upload":false,"outputFile":"/project/assets/sauce-test-report.json","myOption":"test"}',
+    ]);
+  });
+});
+
+describe('normalizeFormat', () => {
+  const assetDir = '/project/assets';
+
+  it('should normalize format with both quoted format type and path', () => {
+    expect(normalizeFormat(`"html":"formatter/report.html"`, assetDir)).toBe(
+      `"html":"/project/assets/formatter/report.html"`,
+    );
+  });
+
+  it('should normalize format with only one pair of quote', () => {
+    expect(normalizeFormat(`"html:formatter/report.html"`, assetDir)).toBe(
+      `"html":"/project/assets/formatter/report.html"`,
+    );
+  });
+
+  it('should normalize format with no quotes', () => {
+    expect(normalizeFormat(`html:formatter/report.html`, assetDir)).toBe(
+      `"html":"/project/assets/formatter/report.html"`,
+    );
+  });
+
+  it('should normalize format with file path type', () => {
+    expect(
+      normalizeFormat(
+        `"file://formatter/implementation":"report.json"`,
+        assetDir,
+      ),
+    ).toBe(`"file://formatter/implementation":"/project/assets/report.json"`);
+  });
+
+  it('should throw an error for an invalid file path type', () => {
+    expect(() => {
+      normalizeFormat(`file://formatter/implementation:report.json`, assetDir);
+    }).toThrow('Invalid format setting detected');
+  });
```

Comment:
I'm a little confused by this. So if the input is not quoted, an error is thrown, but if it is quoted, it gets normalized? That behaviour is so surprising, I was blankly staring at these tests for a minute before I noticed the quotes...

I thought the conclusion of our Friday discussion was we'd just let them use an absolute `file:///` url, but we're ok that they'd get a path error.

Reply:
This gives me something to think about for a while.

This no quote setting is the Cucumber deprecated format, and it's the exact ambiguous colons behavior https://github.com/cucumber/cucumber-js/blob/main/docs/deprecations.md#ambiguous-colons-in-formats
`file://C:\custom\formatter` -> `"file://C:\custom\formatter"`

Given no quotes here,  the current implementation will normalize `file://my_file_path:output_file` to `"file":"/project/asset_dir/my_file_path:output_file"`. It's difficult to debug just by the Cucumber thrown error message ⬇️ And this error is going to be introduced by us 😂 .

```
Error: Cannot find module 'file'
Require stack:
- /Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/lib/formatter/builder.js
- /Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/lib/api/formatters.js
- /Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/lib/api/run_cucumber.js
- /Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/lib/api/index.js
- /Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/lib/cli/index.js
- /Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/lib/cli/run.js
- /Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/bin/cucumber-js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Function.Module._resolveFilename.sharedData.moduleResolveFilenameHook.installedValue [as _resolveFilename] (/Users/chef/bundle/__project__/node_modules/@cspotcode/source-map-support/source-map-support.js:811:30)
    at Function.Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.loadFile (/Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/src/formatter/builder.ts:126:16)
    at Object.loadCustomClass (/Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/src/formatter/builder.ts:111:46)
    at Object.getConstructorByType (/Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/src/formatter/builder.ts:74:32)
    at Object.build (/Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/src/formatter/builder.ts:42:57)
    at initializeFormatter (/Users/chef/bundle/__project__/node_modules/@cucumber/cucumber/src/api/formatters.ts:65:35)
 suite="My Cucumber Test"
```

This PR wants to address ambiguous behavior and aligns with Cucumber's deprecation, there are two ways to resolve the issue: either process the input by splitting at the second colon or block the input entirely. I believe it is more effective to throw an error early, preventing users from encountering the issue and encouraging them to migrate proactively.

For `"file://my_file_path":"output_file"`, the potential issue is the `my_file_path` could be misconfigured with the runner upgrade. But at least this pattern can work for a while.

Reply:
Ah got it, thanks!

Can you make the error message a bit more precise then? `Invalid format setting detected` still doesn't really tell the user what the actual problem is or how to fix it.

Reply:
👍 Sure!